### DISCUSSION TITLE
Supabase Edge Functions Deno 2.0 support

### DISCUSSION THREAD

**Discussion by devidevio** [supabase/supabase#29552] [Feature Requests] [answered]
Just wanted to ask if there are any plans if and when Deno 2.0 will be supported for Supabase Edge Functions.

The latest supported version for Deno is currently v1.45.2 (per latest CLI). Still 7 versions behind (v.146.3) before v2.0.

Thanks in advance.

**Comment by NightTrek** [comment]
I am also eagerly awaitting this as its a requirement in order for us to be able to use some of our dependencies properly

**Comment by AntonOfTheWoods** [comment]
It really, really feels like `supabase` is a "you can do what you want with our source" company rather than an "open source" company. Sigh. They just don't seem to say anything public about upstream versions of anything they use. You know, like of `postgres`... It was completely silent on `postgres` with the "official" version being `15.1` until one of their devs (Sam Rose) started pumping out new versions (`15.8`, `16.3`, even `17` with `orioledb`!). Movement on this didn't appear to start (according to the public branches anyway) until around the time `pg17` dropped (maybe there was a `15.6` version floating around?), so maybe it'll all just arrive when `deno3` drops!

Alas, I couldn't find anything at all about "support" of these new versions, nor what their status was at all, except they are now listed on `supabase/postgres` releases. I'll bet their answer will be "we don't have the bandwidth to communicate on this"...

**Reply by laktek** [reply]
For Deno upgrades within Edge Functios, we announce them in Changelog https://github.com/orgs/supabase/discussions/29189

**Comment by laktek** [comment]
We plan to upgrade to Deno 2 after a Long Term Support (LTS) release (ie. 2.1). This is mainly so that we can guarantee we can offer a stable and a secure service.

Are you missing anything particular from the current version we support (1.45.2)?

**Reply by AntonOfTheWoods** [reply]
The problem from my point of view is that it would be nice - from an open source community perspective - to at least have this sort of thing in the docs (or a blog) rather than, if you're lucky, getting a project maintainer come and reply to a specific request about a particular version. I'm not even expecting for there to be any discussion or voting (like in a non commercial project) - just a push "this is our current thinking and timelines". These sorts of things seem pretty trivial to me - certainly not a huge business expense to have some company docs guru with a blog section...

**Reply by laktek** [reply]
@AntonOfTheWoods Good feedback! yeah, agree we can be more transparent about upcoming releases.

**Reply by laktek** [reply]
As for Edge Functions Deno upgrades, we used to do every minor version after a month of its release. But decided to wait for LTS release of Deno 2, mainly to avoid any surprise bugs. We'll go back to the regular schedule after the 2.1 upgrade, and I will add a section in our docs, linking to release announcements and the timeline.

**Reply by AntonOfTheWoods** [reply]
Now all you have to do is convince the boss ;-).

**Reply by laktek** [reply]
haha! not the boss but the other devs :) Each has its own nuances. Writing an internal message as we speak.

**Reply by adifyr** [reply]
Hi. Deno 2.1 with LTS has [been released](https://deno.com/blog/v2.1).

**Reply by AntonOfTheWoods** [reply]
> Hi. Deno 2.1 with LTS has [been released](https://deno.com/blog/v2.1).

As mentioned above, "As for Edge Functions Deno upgrades, we used to do every minor version *after a month* of its release."

But it's a very major release, so I guess we can be a bit more patient!

**Reply by jmcnulty611** [reply]
Just checking back - any update on Deno 2.x for Edge Functions?

**Reply by janroest** [reply]
we are already near to 2.2 (v2.1.9). So supabase its your turn. 👀

**Reply by laktek** [reply]
We are closing in on the update. Hoping to make an early release within the next 2-4 weeks.

**Comment by laktek** [comment]
UPDATE: We have a PR open for 2.1 upgrade in EdgeRuntime (with all tests passing). https://github.com/supabase/edge-runtime/pull/49

After merging the PR, we will create a preview release of Supabase CLI with Deno 2.1 enabled. It would be great if we could get some early testers to give feedback.

I will give an update again when CLI release is ready.

**Reply by AntonOfTheWoods** [reply]
@laktek , wouldn't a good first step be to publish beta docker images on dockerhub (or ghcr)?

Or is that what https://github.com/supabase/edge-runtime/pkgs/container/edge-runtime/363049202?tag=v1.68.0-develop.5 is (with curious version numbering)?

EDIT: Ok, so yes, it is, but:

```bash
root@transcrobes-supabase-functions-66bbc4899c-fwkmx:/# /usr/local/bin/edge-runtime --version
edge-runtime 0.1.0
deno 2.1.4 (release, x86_64-unknown-linux-gnu)
root@transcrobes-supabase-functions-66bbc4899c-fwkmx:/# /usr/local/bin/edge-runtime start --port 80 --main-service /home/deno/functions/main
Warning Implicitly using latest version (0.224.0) for https://deno.land/std/path/posix/mod.ts
Warning Implicitly using latest version (0.224.0) for https://deno.land/std/http/status.ts
worker boot error: Module not found: file:///var/tmp/sb-compile-edge-runtime/main/index.ts
Error: main worker boot error

Caused by:
    0: worker boot error
    1: Module not found: file:///var/tmp/sb-compile-edge-runtime/main/index.ts
```

So not sure what's going on here. This works fine with `v1.67.2`.

**Reply by laktek** [reply]
@AntonOfTheWoods You can now try it with Supabase CLI https://github.com/orgs/supabase/discussions/34054

**Reply by AntonOfTheWoods** [reply]
Thanks @laktek . I do wish we could have some docs rather than having to reverse engineer from the cli/docker-compose but at I guess I shouldn't complain!

**Comment by laktek** [comment]
We now have a public preview of Deno 2.1 which you can try with local CLI https://github.com/orgs/supabase/discussions/34054

**Reply by nathanaeng** [reply]
When will this be a production-ready option?

**Reply by christophemenager** [reply]
Same question... It has been months since deno 2.1 has been released

**Comment by AntonOfTheWoods** [comment]
The next LTS has now been released...

**Reply by jmcnulty611** [reply]
Watching too.

**Reply by AntonOfTheWoods** [reply]
Actually the next LTS has NOT been released. The Deno team is also an absolute mess - it's not clear they have even decided on an LTS policy and they have contradicting information on what releases will be LTS (2.1 or 2.2?) and the pattern (2.1 is LTS, 2.2 is not LTS, 2.3...? Also, apparently, NOT an LTS).

Both this project and Deno are unfortunately not very well managed projects from a documentation and organisation point of view...

**Comment by jmcnulty611** [comment]
Should the Supabase integration of Deno 2.x (at LTS) be "Edge Functions 2"
and leave current Edge Functions at Deno 1.x

**Comment by laktek** [comment, marked as answer]
All Edge Function regions now run Deno 2.x (2.1.4 is the compatble Deno release). Check the Changelog https://github.com/orgs/supabase/discussions/37941

**Comment by ardabeyazoglu** [comment]
Current LTS version is 2.5 for more than 6 months, any plan to upgrade it ?
